Add `nix run` and `nix build`

srid opened the pull request:
This is largely the work of @MatthewCroughan to enable `nix run` to be used to quickly run the website locally with hot reload, and `nix build` to generate the static website.

This PR should be squash merged, since there are a lot commits.

srid committed:
Move content to ./content

So .nix files, etc. are not copied over to generated website.

srid committed:
Add bin/run script to run locally

srid committed:
index page should have site title

srid commented:
new README view: https://github.com/Plutonomicon/plutonomicon/tree/srid/mv

srid committed:
Add vscode settings & extension recommendations

MatthewCroughan reviewed (changes requested):
@srid 

Can you not add an `apps` attribute to the flake so that you can simply run `nix run .#live` from this flake instead of having to add another file/folder `(bin/run)`to this repo? Would you accept me making this change as a commit on top of what you've done?

srid commented:
> Would you accept me making this change as a commit on top of what you've done?

Yes, thank you, @MatthewCroughan

MatthewCroughan commented:
I've rebased this branch on `main` in order to allow myself to make the change, since the `flake.nix` has been updated in the meantime.

srid commented:
@MatthewCroughan Okay, but please no rebase in future for multi-author branches; you can just `git merge origin/main`. We can always squash merge the PR for cleaner history `main`.

MatthewCroughan committed:
flake: add apps.live

This uses emanote to live serve the website. Use `nix run .#live`

srid commented on flake.nix:
```diff
@@ -17,6 +17,17 @@
+          type = "app";
+          script = pkgs.writers.writeBash "emanotePlutonomiconLiveReload.sh" ''
+            set -xe
+            export PORT=7071
```
Does `nix run` pass the user environment? If so, we would want to use it, and fallback to 7071 only if no $PORT is set. It allows user to change the port number if they desire.

srid replied:
Not critical for this PR though.

MatthewCroughan replied:
`PORT` seems to be a bit general. Can you not make Emanote take `EMANOTE_PORT` instead? I can imagine `PORT` being set by a lot of things, for whatever reason.

srid replied:
> I can imagine `PORT` being set by a lot of things, for whatever reason.

Can you elaborate?

srid replied:
Or better, I should make it take it as argument, along with `$HOST`: `emanote run -b 127.0.0.1:7071`

MatthewCroughan replied:
I've implemented this in e7272cb18da8dc2be50870bc1e3f5dae0b36481f

srid replied:
> Or better, I should make it take it as argument, along with `$HOST`: `emanote run -b 127.0.0.1:7071`

Done in https://github.com/srid/emanote/commit/cb8c989bd39563ad14cb569f1c9e272efd5c5c31

srid commented:
Might as well make `live` the defaultApp, so that `nix run .` works too?

MatthewCroughan committed:
apps.live: allow overriding of port

If EMANOTE_PORT is set, it will set PORT to the value of it. Otherwise, it will use the value 7071

MatthewCroughan committed:
flake: point defaultApp to apps.live

Makes `nix run .` run apps.live

MatthewCroughan commented:
> Might as well make `live` the defaultApp, so that `nix run .` works too?

Done in `e20aada56b40b2bf56e6aa32d620e088e3864a15`

MatthewCroughan committed:
Remove obsolete 'nix run' references

Removes obsolete code and references to our usage of `nix run`

srid committed:
Shorter command to get local server running

srid committed:
`nix run` is enough to get it running

MatthewCroughan committed:
flake: add defaultPackage

srid committed:
`nix build` generates the website now

srid committed:
vscode: disable autoformat

Because nixpkgs-fmt is not available.

L-as commented on .vscode/extensions.json:
```diff
@@ -0,0 +1,14 @@
+	"recommendations": [
+		"yzhang.markdown-all-in-one",
+		"svsool.markdown-memo",
+		"bierner.markdown-checkbox",
```
What is this?

srid replied:
It's https://marketplace.visualstudio.com/items?itemName=bierner.markdown-checkbox

Removed now, along with a few others; they came from https://github.com/srid/emanote-template/tree/master/.vscode

srid committed:
vscode: Remove unnecessary extensions

srid committed:
Revert the moves

srid committed:
Revert .vscode

srid committed:
Restore README instructions

srid committed:
Fix dir

srid committed:
Fix 'nix build'
